docs/monitoring.md

```diff
@@ -159,22 +159,22 @@ Security options for the Spark History Server are covered more detail in the
     <td>false</td>
     <td>
       Indicates whether the history server should use kerberos to login. This is required
-      if the history server is accessing HDFS files on a secure Hadoop cluster. If this is
-      true, it uses the configs <code>spark.history.kerberos.principal</code> and
-      <code>spark.history.kerberos.keytab</code>.
+      if the history server is accessing HDFS files on a secure Hadoop cluster.
     </td>
   </tr>
   <tr>
     <td>spark.history.kerberos.principal</td>
     <td>(none)</td>
     <td>
+      Precondition: <code>spark.history.kerberos.enabled=true</code><br/><br/>
```

Comment:
@HeartSaVioR, seems `Precondition: ` is new style. Can we follow one prevailing style we use in the current code base? Or I actually think it's just better to say "it's effective when spark.xx.xx is enabled." at the end of each description. Current one seems catching too much attention.

Reply:
I'm seeing different styles in current codebase - I've sought three cases and styles are all different:

> spark.dynamicAllocation

if dynamic allocation is enabled (either placed on the first or the last)

> `spark.task.reaper.*`

When <code>spark.task.reaper.enabled = true</code>

> spark.sql.adaptive.shuffle.reducePostShufflePartitions.enabled

When true and '${ADAPTIVE_EXECUTION_ENABLED.key}' is enabled,

Common part across these finding is that we prefer putting it to the first part so that end user can indicate it earlier instead of reading all description and realized it later.

Looks like second and third style look similar, I'll take these style and update here.